interfaceUtils.py

```diff
@@ -10,28 +10,34 @@
 __all__ = ['requestBuildArea', 'runCommand',
            'setBlock', 'getBlock',
            'placeBlockBatched', 'sendBlocks']
+__author__ = "Nils Gawlik <nilsgawlik@gmx.de>"
+__date__ = "11 March 2021"
 # __version__
+__credits__ = "Nils Gawlick for being awesome and creating the framework" + \
+    "Flashing Blinkenlights for general improvements"
 
 import warnings
 
 import requests
 
 
 class Interface():
-    """**Provides tools for interacting with the HTML interface.**
+    """**Provides tools for interacting with the HTML interface**.
 
     All function parameters and returns are in local coordinates.
     """
 
-    def __init__(self, offset=(0, 0, 0), useBatching=True, batchSize=100):
-        self.buffer = []
+    def __init__(self, offset=(0, 0, 0), buffering=False, bufferlimit=4096):
         self.offset = offset
-        self.useBatching = useBatching
-        self.batchSize = batchSize
+        self.__buffering = False
+        self.bufferlimit = 4096
+        self.buffer = []
 
-    def requestBuildArea(self):
-        """**Returns the building area.**"""
+    def __del__(self):
+        self.sendBlocks()
 
+    def requestBuildArea(self):
```

Comment:
This is a change in behavior from the way getBuildArea worked before the interface class. In fact it returns less information, since it doesn't include the y-dimension.
The root problem is that the WorldSlice class takes a tuple, but the http interface returns a dictionary. This is just for historical reasons (WorldSlice is older that the build area functionality in the interface).
If that mismatch is to be fixed, I don't think changing the requestBuildArea functionality to the format of the WorldSlice class is the way to go.

Reply:
I agree it's messy, what do you suggest?
Should we rework that part of WorldSlice?